packages/admin/admin-generator/src/commands/generate/config/transformConfig.ts

```diff
@@ -0,0 +1,611 @@
+import * as ts from "typescript";
+
+const supportedImportPaths = [
+    "[type=grid].columns.filterOperators",
+    "[type=grid].columns.block",
+    "[type=grid].columns.component",
+    // TODO implement in generator "[type=grid].columns.renderCell",
+
+    //support in up to 5 levels of nested fields (eg. fieldSet)
+    ...Array.from(Array(5).keys()).map((i) => `[type=form]${".fields".repeat(i + 1)}.validate`),
+    ...Array.from(Array(5).keys()).map((i) => `[type=form]${".fields".repeat(i + 1)}.block`),
+    ...Array.from(Array(5).keys()).map((i) => `[type=form]${".fields".repeat(i + 1)}.component`),
+];
+const supportedInlineCodePaths = [
+    // TODO implement in generator "[type=grid].columns.filterOperators",
+    "[type=grid].columns.renderCell",
+
+    //support in up to 5 levels of nested fields (eg. fieldSet)
+    ...Array.from(Array(5).keys()).map((i) => `[type=form]${".fields".repeat(i + 1)}.validate`),
+];
+
+// transform the config file to replace all imports and inline code with a { code, imports } object
+// this is needed to be able to execute the config file in a node environment
+export function transformConfigFile(fileName: string, sourceText: string) {
+    const sourceFile = ts.createSourceFile(
+        fileName,
+        sourceText,
+        ts.ScriptTarget.Latest, // language version
```

Comment:
In theory the language version could be different in a project. Is this something we should consider? Or should we always assume the package's version?

Reply:
[78769e2](https://github.com/vivid-planet/comet/pull/3781/commits/78769e20e8b747edc1303176314e649de53f427e)